dev-java/jaxen/jaxen-1.2.0-r1.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Skeleton command:
+# java-ebuilder --generate-ebuild --workdir . --pom pom.xml --download-uri https://github.com/jaxen-xpath/jaxen/archive/refs/tags/v1.2.0.tar.gz --slot 1.2 --keywords "~amd64 ~arm ~arm64 ~ppc64 ~x86" --ebuild jaxen-1.2.0.ebuild
+
+EAPI=7
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="jaxen:jaxen:1.2.0"
+JAVA_TESTING_FRAMEWORKS="junit"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="Jaxen is a universal XPath engine for Java."
+HOMEPAGE="http://www.cafeconleche.org/jaxen/"
+SRC_URI="https://github.com/${PN}-xpath/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="1.2"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="dom4j"
+
+# It seems that most tests depend on dom4j, jdom and xom which all depend on jaxen
+RESTRICT="test"
+
+# Compile dependencies
+# POM: pom.xml
+# xerces:xercesImpl:2.6.2 -> >=dev-java/xerces-2.12.0:2
+# xml-apis:xml-apis:1.3.02 -> >=dev-java/xml-commons-external-1.4.01:1.4
+# POM: pom.xml
+# test? junit:junit:3.8.2 -> >=dev-java/junit-3.8.2:0
+
+DEPEND="
+	dev-java/xerces:2
+	>=virtual/jdk-1.8:*
+	dom4j? ( dev-java/dom4j:1 )
+"
+
+RDEPEND="
+	>=virtual/jre-1.8:*
+"
+
+S="${WORKDIR}"
+
+DOCS=( "${P}"/{LICENSE.txt,README.md} )
+
+JAVA_CLASSPATH_EXTRA="xerces-2"
+JAVA_SRC_DIR="${P}/src/java/main"
```

Comment:
tests are in `IUSE` but settings for tests is missing here.

Reply:
@fordfrog Test are still `RESTRICT="test", just like in `jaxen-1.2.0.ebuild`.
But, how do you think about using the `test` flag instead of introducing the local `dom4j` flag?

So far I only unveiled the dom4j directory which is needed for commons-jelly.
But should (to be tested) also work for `jdom` and `xom`.

Reply:
it came to my mind that we could provide all three use flags. and depending on enabled use flags appropriate tests might be run, if they are not mixed together. if they are mixed together, then all three flags would be needed for tests, as otherwise there is nothing to test. what do you think?

Reply:
My preference? Keep it simple and stupid, similar to what's proposed by java-ebuilder: [jaxen-1.2.0.ebuild.gz](https://github.com/gentoo/gentoo/files/6329290/jaxen-1.2.0.ebuild.gz)
I would like to replace the `dom4j` USE flag with the `test` flag and keep `jdom` and `xom` hidden at the moment.
Unhiding `jdom` and `xom` and enabling tests could be done later.  I know that's kind of misusing the `test` flag.

The other topic is the circular dependency. Presently `dom4j` pulls jaxen in the older slot.  Once you pushed `dom4j` to pulling `jaxen:1.2` ...
But the circular dependency is given upstream and should be solved there.

Reply:
but `java-ebuilder` does not remove any code. and you cannot depend on `test` use flag from `commons-jelly`.

Reply:
> but `java-ebuilder` does not remove any code. and you cannot depend on `test` use flag from `commons-jelly`.

So it's not allowed to have `commons-jelly` with
```
dev-java/jaxen:1.2[test]
```
??
If so, could you please take it as is (Please check the flag description in `metadata.xml`) and let's keep `xom` and `jdom` for later.
The revbump is needed, right?

Reply:
`dev-java/jaxen:1.2[test]` misuses `[test]` for something it's not intended for. i'm fine with `dev-java/jaxen:1.2[dom4j]`, i would not be even afraid to extend it with `[xom,jdom]`, as that might be needed in the future anyway. and portage has some useful hints when these cyclical deps occur. but we can leave it as it is now, for now. there is only one think i don't like about this ebuild and i did not notice it on our previous merge, and that is the fact that it has `[test]` use flag but if the restriction would be removed, the tests would not work as they are not defined. so i'd suggest to remove `[test]` completely as this is just a partial non-working attempt for implementation and the flag is useless without test setup in the ebuild. it still would be useful to keep there the note about why tests are not implemented for future reference.

Reply:
>  so i'd suggest to remove `[test]` completely

Could you please do that on merge?  Or is there any other important changes needed?  
I'm glad to have a solution for `commons-jelly`

Reply:
ok, i'll take it as it is, will clean it up and do final testing.

Reply:
Thanks